aws/terminator/networking.py

```diff
@@ -414,8 +414,23 @@ def name(self):
     def ignore(self):
         return self.name == 'default' or self.name.startswith('default_elb_')
 
+    def revoke_sg_rules(self):
+        # Revoke Egress rules
+        self.client.revoke_security_group_egress(
+            GroupId=self.id, IpPermissions=self.instance.get("IpPermissionsEgress")
+        )
+
+        # Revoke Ingress rules
+        self.client.revoke_security_group_ingress(
+            GroupId=self.id, IpPermissions=self.instance.get("IpPermissions")
+        )
+
     def terminate(self):
-        self.client.delete_security_group(GroupId=self.id)
+        try:
```

Comment:
What happen if we just call the two methods without the if condition? This way we avoid the risk of breakage if the format of the exception change in the future. I imagine we may need to wrap `revoke_sg_rules()` in `try/except` block thus.
```
self.revoke_sg_rules()
self.client.delete_security_group(GroupId=self.id)
```

Reply:
This was my first idea, but we want to cut down on unnecessary requests